nano/core_test/active_transactions.cpp

```diff
@@ -114,6 +124,123 @@ TEST (active_transactions, adjusted_difficulty_priority)
+{
+	nano::system system;
+	nano::node_config node_config (24000, system.logging);
+	node_config.enable_voting = false;
+	node_config.frontiers_confirmation = nano::frontiers_confirmation_mode::disabled;
+	auto & node1 = *system.add_node (node_config);
+	nano::genesis genesis;
+	nano::keypair key1, key2, key3;
+
+	auto send1 (std::make_shared<nano::state_block> (nano::test_genesis_key.pub, genesis.hash (), nano::test_genesis_key.pub, nano::genesis_amount - 10 * nano::xrb_ratio, key1.pub, nano::test_genesis_key.prv, nano::test_genesis_key.pub, system.work.generate (genesis.hash ())));
+	auto send2 (std::make_shared<nano::state_block> (nano::test_genesis_key.pub, send1->hash (), nano::test_genesis_key.pub, nano::genesis_amount - 20 * nano::xrb_ratio, key2.pub, nano::test_genesis_key.prv, nano::test_genesis_key.pub, system.work.generate (send1->hash ())));
+	auto open1 (std::make_shared<nano::state_block> (key1.pub, 0, key1.pub, 10 * nano::xrb_ratio, send1->hash (), key1.prv, key1.pub, system.work.generate (key1.pub)));
+	auto open2 (std::make_shared<nano::state_block> (key2.pub, 0, key2.pub, 10 * nano::xrb_ratio, send2->hash (), key2.prv, key2.pub, system.work.generate (key2.pub)));
+	auto send3 (std::make_shared<nano::state_block> (key2.pub, open2->hash (), key2.pub, 9 * nano::xrb_ratio, key3.pub, key2.prv, key2.pub, system.work.generate (open2->hash ())));
+	node1.process_active (send1); // genesis
+	node1.process_active (send2); // genesis
+	node1.process_active (open1); // key1
+	node1.process_active (open2); // key2
+	node1.process_active (send3); // key2
+	system.deadline_set (10s);
+	while (node1.active.size () != 5)
+	{
+		ASSERT_NO_ERROR (system.poll ());
+	}
+
+	{
+		std::lock_guard<std::mutex> active_guard (node1.active.mutex);
+		// Update difficulty to minimum
+		auto send1_root (node1.active.roots.find (send1->qualified_root ()));
+		auto send2_root (node1.active.roots.find (send2->qualified_root ()));
+		auto open1_root (node1.active.roots.find (open1->qualified_root ()));
+		auto open2_root (node1.active.roots.find (open2->qualified_root ()));
+		auto send3_root (node1.active.roots.find (send3->qualified_root ()));
+		node1.active.roots.modify (send1_root, [](nano::conflict_info & info_a) {
+			info_a.difficulty = std::numeric_limits<std::uint64_t>::min () + 1;
+		});
+		node1.active.roots.modify (send2_root, [](nano::conflict_info & info_a) {
+			info_a.difficulty = std::numeric_limits<std::uint64_t>::min () + 1;
+		});
+		node1.active.roots.modify (open1_root, [](nano::conflict_info & info_a) {
+			info_a.difficulty = std::numeric_limits<std::uint64_t>::min () + 1;
+		});
+		node1.active.roots.modify (open2_root, [](nano::conflict_info & info_a) {
+			info_a.difficulty = std::numeric_limits<std::uint64_t>::min () + 1;
+		});
+		node1.active.roots.modify (send3_root, [](nano::conflict_info & info_a) {
+			info_a.difficulty = std::numeric_limits<std::uint64_t>::min () + 1;
+		});
```

Comment:
These `modify` calls are very similar, could reduce some duplication by wrapping the whole thing in a local lambda like:
```
auto modify_difficulty = [&roots = node1.active.root](auto & root) {
	roots.modify (root, [](nano::conflict_info & info_a) {
		info_a.difficulty = std::numeric_limits<std::uint64_t>::min () + 1;
	});
};

modify_difficulty (send1_root);
....
```
Or put it in a function, or perhaps just extract the callback lambda in modify function call (and make a local variable to `roots`) would be fine too :).

Same can be done in the other newly added test.

Reply:
Updated